tools/serve/serve.py

```diff
@@ -845,6 +846,61 @@ def start_quic_transport_server(logger, host, port, paths, routes, bind_address,
         startup_failed(logger)
 
 
+class WebTransportH3Daemon(object):
+    def __init__(self, host, port, handlers_path=None, private_key=None, certificate=None, log_level=None):
+        args = ["python3", "wpt", "serve-webtransport-h3"]
```

Comment:
I changed to run the server in-process. The server takes a logging object to integrate log settings.

Unfortunately this brings another issue. serve.py now depends on webtransport_h3_server.py which depends on aioquic. I'm not sure how to handle the dependency problem. Options would be (1) add aioquic dependency to `tools/serve`, `tools/wptrunner`, `docs` etc by default, (2) revert the change and keep using virtualenv+Popen, passing logging setting. Could you give me some advice?

Reply:
I see that there's already both aioquic in `tools/quic/requirements.txt` and a "vendored" copy in tools/third_party/aioquic. Does the copy in tools/third_party/aioquic actually work, or is it unused?

Since dependencies via requirements.txt are easier to understand and upgrade, I'm thinking that if we could use that mechanism, that would be nice. I've learned from @jgraham that Gecko CI doesn't use those dependencies at all, but if we could ensure that the dependency is only used in tests that require some flag (`--webtransport-h3`?) to run, then it would be fine in practice.

Reply:
As far as I know the copy in tools/third_party/aioquic is not used. Maybe @yutakahirano can confirm. I'm happy to remove it in a follow-up PR if nobody uses it.

I agree that using requirements.txt and virtualenv is easier to maintain dependencies. I re-add commands.json and the server now runs in a separate process. Log level is passed from serve.py so that serve.py and the server use the same log level. Does this make sense?

Yes, as far as I understand the dependency is only used in tests that require `--webtransport-h3` to run. Please let me know if I miss something. (I may miss something actually, since I'm not very familiar with wpt infrastructure)

The current implementation of the server can run as a in-proc threaded server if aioquic dependency is allowed. It should be able to integrate into serve.py in the future.

Reply:
There's no need to clean up the duplicate aioquic dependency in this PR, if you get it to work that's fine.

Here, I still think it would be good to change the implementation to an in-proc threaded server. If the code path that imports aioquic is hit even when not running those tests, then requiring the `--webtransport-h3` argument for that code path sounds like a good fix. It could be made unconditional at a later point when it's enabled everywhere and our dependency situation is a bit more reasonable...

Reply:
Sorry I should have clarified that, IIUC, we can't make it an in-proc threaded server if we use virtualenv+requirements.txt to handle aioquic dependency (if I missed something, please let me know).

My understanding is that there are two options:
* Update tools/third_party/aioquic and depend on it. This enables the server runs in-proc.
* Use virtualenv and run the server in a separate process via commands.json (the current PR).

I think either is fine for us (Chromium). Or there might be a better way to handle aioquic dependency.